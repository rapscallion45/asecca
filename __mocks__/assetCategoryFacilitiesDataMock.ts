import { IAssetCategoryFacilitiesDataPayload } from '@/lib/api/api-types';

/**
 * mock file defining returned data from API GET /collection/asset_category/api/facilities
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.16
 */
const assetCategoryFacilitiesDataMock: IAssetCategoryFacilitiesDataPayload = [
  'ASECCA',
  'Techsel',
];
export default assetCategoryFacilitiesDataMock;
