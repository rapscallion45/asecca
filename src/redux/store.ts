import { configureStore, Action } from '@reduxjs/toolkit';
import { ThunkAction } from 'redux-thunk';
import logger from 'redux-logger';
import userPermissionReducer from './slices/userPermissionSlice';
import costsConfigReducer from './slices/costsConfigSlice';
import notificationsReducer from './slices/notificationsSlice';
import themeReducer from './slices/themeSlice';
import { collectionsKanbanSlice } from './slices/kanbanSlice';

/**
 * Configuration of the redux store to manage the global application state
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @see See [more info on Redux Store object](https://redux.js.org/api/store)
 *
 * @constant
 * @type {ToolkitStore<Object>}
 */
const store = configureStore({
  reducer: {
    userPermission: userPermissionReducer,
    costsConfig: costsConfigReducer,
    notifications: notificationsReducer,
    theme: themeReducer,
    collectionsKanban: collectionsKanbanSlice.reducer,
  },
  // @ts-ignore
  middleware:
    process.env.NODE_ENV === 'development'
      ? (getDefaultMiddleware) => getDefaultMiddleware().concat(logger)
      : (getDefaultMiddleware) => getDefaultMiddleware(),
});

/**
 * Shorthand application store dispatch type helper
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @typedef AppDispatch - application specific store dispatch type
 */
export type AppDispatch = typeof store.dispatch;

/**
 * Shorthand application state type helper
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @typedef AppState - application specific state object shape
 */
export type AppState = ReturnType<typeof store.getState>;

/**
 * Shorthand application Thunk Action type helper
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @typedef AppThunk - application specific Thunk action definition
 */
export type AppThunk = ThunkAction<void, AppState, unknown, Action<string>>;

export default store;
