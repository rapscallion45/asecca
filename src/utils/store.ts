/* redux store configuration placeholder */
const useStore = () => {};
export default useStore;
