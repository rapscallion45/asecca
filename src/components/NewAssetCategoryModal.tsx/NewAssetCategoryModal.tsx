import { FC, useState, useCallback, ChangeEvent, useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { AppDispatch } from '@/redux/store';
import {
  Box,
  Button,
  TextField,
  RadioGroup,
  FormControlLabel,
  Radio,
  FormLabel,
  FormControl,
  Stack,
  Skeleton,
  Grid,
  Typography,
  Checkbox,
} from '@mui/material';
import { LoadingButton } from '@mui/lab';
import { addNotification } from '@/redux/slices/notificationsSlice';
import {
  IAssetCategoryFacilitiesDataPayload,
  INewAssetCategoryData,
  INewAssetCategoryDataPayload,
} from '@/lib/api/api-types';
import assetCategoryService from '@/services/assetCategory/assetCategory';
import assetCategoryFacilitiesDataMock from '../../../__mocks__/assetCategoryFacilitiesDataMock';

/**
 * New Asset Category Modal Props
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.16
 *
 * @typedef INewAssetCategoryModalProps
 * @prop {any} closeModal - on close modal callback handler
 */
interface INewAssetCategoryModalProps {
  closeModal?: (fetchNewTypes: boolean) => void;
}

/**
 * New Asset Category Modal
 *
 * Presents the Collection Form New Asset modal to the user, populated with data
 * fetched from API: /api/collection/asset_category/api/facilities
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.16
 *
 * @component
 * @param {INewAssetCategoryModalProps} props - component props
 * @returns {FC} - collection form new asset modal functional component
 */
const NewAssetCategoryModal: FC<INewAssetCategoryModalProps> = (props) => {
  const { closeModal } = props;
  const dispatch = useDispatch<AppDispatch>();

  /**
   * Local state data for asset category facilities loading flag
   *
   * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
   * @since 0.0.16
   *
   * @constant
   */
  const [loading, setLoading] = useState<boolean>(true);

  /**
   * Local state data for new asset category saving flag
   *
   * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
   * @since 0.0.16
   *
   * @constant
   */
  const [saving, setSaving] = useState<boolean>(false);

  /**
   * Local state data for new asset category facilities list
   *
   * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
   * @since 0.0.16
   *
   * @constant
   */
  const [facilities, setFacilities] =
    useState<IAssetCategoryFacilitiesDataPayload>(
      assetCategoryFacilitiesDataMock
    );

  /**
   * Local state data for creation of the new asset category
   *
   * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
   * @since 0.0.16
   *
   * @constant
   */
  const [newAssetData, setNewAssetData] = useState<INewAssetCategoryData>({
    name: '',
    co2: 0,
    data_bearing: false,
    removable_storage: false,
    serialized: false,
    compatible_facilities: assetCategoryFacilitiesDataMock,
  });

  /* on first load, fetch the asset category facility types */
  useEffect(() => {
    /**
     * Handles the fetching of the asset category facilities data
     *
     * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
     * @since 0.0.16
     *
     * @method
     */
    const fetchFacilities = async () => {
      /* indicate loading in progress */
      setLoading(true);

      /* call API */
      const res = await assetCategoryService.getAssetCategoryFacilities();

      /* clear loading in progress flag */
      setLoading(false);

      /* add a notification and clear local state if bad response from server */
      if (res.status !== 200) {
        dispatch(
          addNotification({
            message: `Failed to load Asset Category Facilities: ${res.statusText}`,
            variant: 'error',
          })
        );
        // setFacilities([]);
        return;
      }

      /* good response, set facilitiy state data */
      const data: IAssetCategoryFacilitiesDataPayload = await res.json();
      setFacilities(data);
    };

    fetchFacilities();
  }, [dispatch]);

  /**
   * Handles the editing of the form data
   *
   * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
   * @since 0.0.16
   *
   * @method
   * @param {string | boolean} value - form value to be updated
   * @param {string} itemKey - form item key to be updated
   */
  const handleEdit = (
    value: string | Array<string | null> | boolean,
    itemKey: string
  ) => {
    setNewAssetData({ ...newAssetData, [itemKey]: value });
  };

  /**
   * Handles the cancel and closure of the modal
   *
   * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
   * @since 0.0.16
   *
   * @method
   */
  const handleCancel = useCallback(() => {
    if (closeModal) closeModal(false);
  }, [closeModal]);

  /**
   * Handles the saving of the form data
   *
   * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
   * @since 0.0.16
   *
   * @method
   */
  const handleSave = useCallback(async () => {
    /* indicate save in progress */
    setSaving(true);

    /* call API */
    const res = await assetCategoryService.setNewAssetCategory(
      newAssetData as INewAssetCategoryDataPayload
    );

    /* clear save in progress flag */
    setSaving(false);

    /* add a notification and reject if bad response from server */
    if (res.status !== 200) {
      dispatch(
        addNotification({
          message: `Failed to save New Asset Category: ${res.statusText}`,
          variant: 'error',
        })
      );
      return;
    }

    /* indicate successful save of New Asset Category */
    dispatch(
      addNotification({
        message: `Successfully saved New Asset Category: ${res.statusText}`,
        variant: 'success',
      })
    );

    /* good response, close the modal as we're done */
    if (closeModal) closeModal(true);
  }, [newAssetData, closeModal, dispatch]);

  /**
   * Handles the saving of the form data
   *
   * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
   * @since 0.0.16
   *
   * @method
   * @param {string | null} facility - facility name
   * @param {boolean} selected - whether user has selected the facilities checkbox
   * @returns {Array<string>} - updated compatible facilities string array
   */
  const getCompatibleFacilitiesUpdateValue = (
    facility: string | null,
    selected: boolean
  ): Array<string | null> =>
    selected
      ? newAssetData.compatible_facilities.concat(facility)
      : newAssetData.compatible_facilities.filter((item) => item !== facility);

  return !loading ? (
    <Box>
      <Box
        component="form"
        sx={{
          '& .MuiTextField-root': { my: 1 },
        }}
        noValidate
        autoComplete="off"
        display="flex"
        flexDirection="column"
      >
        <Box display="flex" flexDirection="column">
          <TextField
            id="new-asset-name-input"
            label="Name"
            value={newAssetData.name}
            onChange={(event: ChangeEvent<HTMLInputElement>) => {
              handleEdit(event.target.value, 'name');
            }}
            placeholder="New Asset Category Name..."
          />
          <TextField
            id="new-asset-co2-input"
            label="CO2"
            value={newAssetData.co2}
            onChange={(event: ChangeEvent<HTMLInputElement>) => {
              handleEdit(event.target.value, 'co2');
            }}
            placeholder="New Asset Category CO2 value..."
          />
          <FormControl sx={{ mt: 1 }}>
            <FormLabel id="data-bearing-radio-buttons-group-label">
              Data Bearing:
            </FormLabel>
            <RadioGroup
              aria-labelledby="data-bearing-radio-buttons-group-label"
              name="data-bearing-radio-buttons-group"
              value={newAssetData.data_bearing ? 'true' : 'false'}
              onChange={(event: ChangeEvent<HTMLInputElement>) => {
                handleEdit(event.target.value === 'true', 'data_bearing');
              }}
              row
            >
              <FormControlLabel
                value="true"
                control={<Radio color="secondary" />}
                label="True"
              />
              <FormControlLabel
                value="false"
                control={<Radio color="secondary" />}
                label="False"
              />
            </RadioGroup>
          </FormControl>
          <FormControl sx={{ mt: 1 }}>
            <FormLabel id="removable-storage-radio-buttons-group-label">
              Removable Storage:
            </FormLabel>
            <RadioGroup
              aria-labelledby="removable-storage-radio-buttons-group-label"
              name="removable-storage-radio-buttons-group"
              value={newAssetData.removable_storage ? 'true' : 'false'}
              onChange={(event: ChangeEvent<HTMLInputElement>) => {
                handleEdit(event.target.value === 'true', 'removable_storage');
              }}
              row
            >
              <FormControlLabel
                value="true"
                control={<Radio color="secondary" />}
                label="True"
              />
              <FormControlLabel
                value="false"
                control={<Radio color="secondary" />}
                label="False"
              />
            </RadioGroup>
          </FormControl>
          <FormControl sx={{ mt: 1 }}>
            <FormLabel id="serialised-radio-buttons-group-label">
              Serialised:
            </FormLabel>
            <RadioGroup
              aria-labelledby="serialised-radio-buttons-group-label"
              name="serialised-radio-buttons-group"
              value={newAssetData.serialized ? 'true' : 'false'}
              onChange={(event: ChangeEvent<HTMLInputElement>) => {
                handleEdit(event.target.value === 'true', 'serialized');
              }}
              row
            >
              <FormControlLabel
                value="true"
                control={<Radio color="secondary" />}
                label="True"
              />
              <FormControlLabel
                value="false"
                control={<Radio color="secondary" />}
                label="False"
              />
            </RadioGroup>
          </FormControl>
          <Box mt={2}>
            <Typography color="text.secondary" mb={1}>
              Compatible Facilities:
            </Typography>
            {facilities.map((facility: string | null) => {
              if (facility)
                return (
                  <Box key={facility} display="flex">
                    <Grid
                      container
                      rowSpacing={1}
                      justifyContent="left"
                      alignItems="center"
                    >
                      <Grid item xs={6} md={6}>
                        <Typography>{facility}:</Typography>
                      </Grid>
                      <Grid item xs={6}>
                        <Checkbox
                          checked={newAssetData.compatible_facilities.some(
                            (item) => item === facility
                          )}
                          onChange={(
                            event: ChangeEvent<HTMLInputElement>,
                            checked: boolean
                          ) =>
                            handleEdit(
                              getCompatibleFacilitiesUpdateValue(
                                facility,
                                checked
                              ),
                              'compatible_facilities'
                            )
                          }
                          color="secondary"
                        />
                      </Grid>
                    </Grid>
                  </Box>
                );
              return null;
            })}
          </Box>
        </Box>
      </Box>
      <Box
        sx={{
          marginTop: 4,
          display: 'flex',
          flexDirection: 'row',
          alignItems: 'center',
          justifyContent: 'right',
        }}
      >
        <Button
          color="secondary"
          variant="outlined"
          onClick={handleCancel}
          disabled={saving}
          sx={{ backgroundColor: 'common.white' }}
        >
          Cancel
        </Button>
        <LoadingButton
          color="secondary"
          variant="contained"
          onClick={handleSave}
          disabled={saving || Boolean(!newAssetData.name)}
          loading={saving}
          sx={{ ml: 2 }}
        >
          Create
        </LoadingButton>
      </Box>
    </Box>
  ) : (
    <Stack spacing={1}>
      {/* display loading skeleton */}
      <Skeleton variant="rectangular" width={310} height={60} />
      <Skeleton variant="rectangular" width={310} height={60} />
      <Skeleton variant="rectangular" width={210} height={30} />
      <Skeleton variant="rounded" width={210} height={60} />
      <Skeleton variant="rectangular" width={210} height={30} />
      <Skeleton variant="rounded" width={210} height={60} />
      <Skeleton variant="rectangular" width={210} height={30} />
      <Skeleton variant="rounded" width={210} height={60} />
      <Skeleton variant="rectangular" width={210} height={30} />
      <Skeleton variant="rounded" width={210} height={60} />
    </Stack>
  );
};

export default NewAssetCategoryModal;
