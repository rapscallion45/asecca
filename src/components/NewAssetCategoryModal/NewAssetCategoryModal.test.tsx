import React from 'react';
import { Provider } from 'react-redux';
import {
  fireEvent,
  render,
  screen,
  waitFor,
  within,
} from '@testing-library/react';
import store from '@/redux/store';
import NewAssetCategoryModal from './NewAssetCategoryModal';
import assetCategoryFacilitiesDataMock from '../../../__mocks__/assetCategoryFacilitiesDataMock';

/* mock save callback */
const mockSaveCallback = jest.fn(() => {});

/* mock close modal callback */
const mockCloseModalCallback = jest.fn(() => {});

/**
 * New Asset Category Modal Unit Tests
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.23
 */
describe('New Asset Category Modal', () => {
  it('Renders correctly', async () => {
    /** Arrange */
    /** perform snapshot test */
    const tree = render(
      <Provider store={store}>
        <NewAssetCategoryModal
          saving={false}
          closeModal={mockCloseModalCallback}
          handleSaveCallback={mockSaveCallback}
        />
      </Provider>
    );
    expect(tree).toMatchSnapshot();
  });

  describe('Form Fields', () => {
    it('Should render all required New Asset Category data fields', async () => {
      /** Arrange */
      /** Act - render the test components */
      render(
        <Provider store={store}>
          <NewAssetCategoryModal
            saving={false}
            closeModal={mockCloseModalCallback}
            handleSaveCallback={mockSaveCallback}
          />
        </Provider>
      );

      /** Assert - ensure all fields are present */
      await waitFor(() => {
        expect(screen.getByLabelText('Name')).toBeInTheDocument();
      });
      expect(screen.getByLabelText('CO2')).toBeInTheDocument();
      expect(screen.getByLabelText('Data Bearing:')).toBeInTheDocument();
      expect(screen.getByLabelText('Removable Storage:')).toBeInTheDocument();
      expect(screen.getByLabelText('Serialised:')).toBeInTheDocument();
      expect(screen.getByText('Compatible Facilities:')).toBeInTheDocument();
      assetCategoryFacilitiesDataMock.forEach((facility: string) =>
        expect(screen.getByText(`${facility}:`)).toBeInTheDocument()
      );
    });
  });

  describe('Form Callbacks', () => {
    beforeAll(() => jest.spyOn(window, 'fetch'));

    it('Should call save callback when values updated and Create button clicked', async () => {
      /** Arrange */
      const testName: string = 'Test Asset Category';
      const testCO2: string = '30';
      const testDataBearing: boolean = true;
      const testRemovableStorage: boolean = true;
      const testSerialised: boolean = true;
      const testCompatibleFacilities: Array<string> = [
        assetCategoryFacilitiesDataMock[1],
      ];

      /** Act - render the test components */
      render(
        <Provider store={store}>
          <NewAssetCategoryModal
            saving={false}
            closeModal={mockCloseModalCallback}
            handleSaveCallback={mockSaveCallback}
          />
        </Provider>
      );

      /** Assert - check we have correct values rendered */
      await waitFor(() => {
        expect(screen.getByLabelText('Name')).toBeInTheDocument();
      });
      assetCategoryFacilitiesDataMock.forEach((facility: string) =>
        expect(screen.getByText(`${facility}:`)).toBeInTheDocument()
      );
      expect(screen.getByText('Create')).toHaveAttribute('disabled');

      /** Act - update field values */
      fireEvent(
        screen.getByLabelText('Name'),
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        })
      );
      fireEvent.change(
        screen.getByPlaceholderText('New Asset Category Name...'),
        {
          target: { value: testName },
        }
      );
      fireEvent(
        screen.getByLabelText('CO2'),
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        })
      );
      fireEvent.change(
        screen.getByPlaceholderText('New Asset Category CO2 value...'),
        {
          target: { value: testCO2 },
        }
      );
      if (testDataBearing)
        fireEvent(
          within(screen.getByLabelText('Data Bearing:')).getByLabelText('True'),
          new MouseEvent('click', {
            bubbles: true,
            cancelable: true,
          })
        );
      if (testRemovableStorage)
        fireEvent(
          within(screen.getByLabelText('Removable Storage:')).getByLabelText(
            'True'
          ),
          new MouseEvent('click', {
            bubbles: true,
            cancelable: true,
          })
        );
      if (testSerialised)
        fireEvent(
          within(screen.getByLabelText('Serialised:')).getByLabelText('True'),
          new MouseEvent('click', {
            bubbles: true,
            cancelable: true,
          })
        );
      fireEvent(
        screen.getByTestId(`${assetCategoryFacilitiesDataMock[0]}-checkbox`),
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        })
      );

      /** Assert - ensure updates have been processed */
      await waitFor(() => {
        expect(screen.getByDisplayValue(testName)).toBeInTheDocument();
      });
      expect(screen.getByDisplayValue(testCO2)).toBeInTheDocument();

      /** Act - click Create button */
      fireEvent(
        screen.getByText('Create'),
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        })
      );

      /** Assert - save callback to have been called with data from user input */
      await waitFor(() => {
        expect(mockSaveCallback).toHaveBeenCalledTimes(1);
      });
      expect(mockSaveCallback).toHaveBeenCalledWith({
        co2: testCO2,
        compatible_facilities: testCompatibleFacilities,
        data_bearing: testDataBearing,
        name: testName,
        removable_storage: testRemovableStorage,
        serialized: testSerialised,
      });
    });

    it('Should call close modal callback when Cancel button clicked', async () => {
      /** Arrange */
      /** Act - render the test components */
      render(
        <Provider store={store}>
          <NewAssetCategoryModal
            saving={false}
            closeModal={mockCloseModalCallback}
            handleSaveCallback={mockSaveCallback}
          />
        </Provider>
      );

      /** Assert - check we have correct values rendered */
      await waitFor(() => {
        expect(screen.getByLabelText('Name')).toBeInTheDocument();
      });
      assetCategoryFacilitiesDataMock.forEach((facility: string) =>
        expect(screen.getByText(`${facility}:`)).toBeInTheDocument()
      );
      expect(screen.getByText('Create')).toHaveAttribute('disabled');

      /** Act - click Cancel button */
      fireEvent(
        screen.getByText('Cancel'),
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        })
      );

      /** Assert - close modal callback to have been called  */
      await waitFor(() => {
        expect(mockCloseModalCallback).toHaveBeenCalledTimes(1);
      });
    });
  });

  describe('Error and Loading', () => {
    /** ensure mock function calls are cleared after test */
    afterEach(() => {
      jest.clearAllMocks();
    });

    it('Should render loading skeleton when loading prop set', async () => {
      /** Arrange */
      /** Act */
      render(
        <Provider store={store}>
          <NewAssetCategoryModal
            saving={false}
            closeModal={mockCloseModalCallback}
            handleSaveCallback={mockSaveCallback}
          />
        </Provider>
      );

      /** Assert - loading skeleton should be rendered while API fetches data */
      await waitFor(() => {
        expect(
          screen.getByTestId('new-asset-category-modal-loading-skeleton')
        ).toBeInTheDocument();
      });
    });
  });
});
