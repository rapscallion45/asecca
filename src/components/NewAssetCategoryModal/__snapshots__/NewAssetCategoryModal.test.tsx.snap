// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Asset Category Modal Renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiStack-root css-1okj3ks-MuiStack-root"
        data-testid="new-asset-category-modal-loading-skeleton"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-58ftfq-MuiSkeleton-root"
          style="width: 310px; height: 60px;"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-58ftfq-MuiSkeleton-root"
          style="width: 310px; height: 60px;"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-58ftfq-MuiSkeleton-root"
          style="width: 210px; height: 30px;"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse css-1ukovu3-MuiSkeleton-root"
          style="width: 210px; height: 60px;"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-58ftfq-MuiSkeleton-root"
          style="width: 210px; height: 30px;"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse css-1ukovu3-MuiSkeleton-root"
          style="width: 210px; height: 60px;"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-58ftfq-MuiSkeleton-root"
          style="width: 210px; height: 30px;"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse css-1ukovu3-MuiSkeleton-root"
          style="width: 210px; height: 60px;"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-58ftfq-MuiSkeleton-root"
          style="width: 210px; height: 30px;"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse css-1ukovu3-MuiSkeleton-root"
          style="width: 210px; height: 60px;"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiStack-root css-1okj3ks-MuiStack-root"
      data-testid="new-asset-category-modal-loading-skeleton"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-58ftfq-MuiSkeleton-root"
        style="width: 310px; height: 60px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-58ftfq-MuiSkeleton-root"
        style="width: 310px; height: 60px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-58ftfq-MuiSkeleton-root"
        style="width: 210px; height: 30px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse css-1ukovu3-MuiSkeleton-root"
        style="width: 210px; height: 60px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-58ftfq-MuiSkeleton-root"
        style="width: 210px; height: 30px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse css-1ukovu3-MuiSkeleton-root"
        style="width: 210px; height: 60px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-58ftfq-MuiSkeleton-root"
        style="width: 210px; height: 30px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse css-1ukovu3-MuiSkeleton-root"
        style="width: 210px; height: 60px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-58ftfq-MuiSkeleton-root"
        style="width: 210px; height: 30px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse css-1ukovu3-MuiSkeleton-root"
        style="width: 210px; height: 60px;"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
